apollo-runtime/src/main/java/com/apollographql/apollo/internal/interceptor/ApolloCacheInterceptor.java

```diff
@@ -135,7 +135,7 @@ InterceptorResponse resolveFromCache(InterceptorRequest request) throws ApolloEx
         }
     );
 
-    if (!records.isPresent()) {
+    if (!records.isPresent() || (networkResponse.parsedResponse.isPresent() && networkResponse.parsedResponse.get().hasErrors())) {
```

Comment:
I think we can move this to the top of `cacheResponse` function
```
    if (networkResponse.parsedResponse.isPresent() && networkResponse.parsedResponse.get().hasErrors()) {
      return Collections.emptySet();
    }
    final Optional<List<Record>> records = networkResponse.cacheRecords.map(
    ....
```

Reply:
hmm im down with short circuiting. will update